src/terminal/input/terminalInput.cpp

```diff
@@ -551,20 +551,23 @@ bool TerminalInput::HandleKey(const IInputEvent* const pInEvent)
         return true;
     }
 
+    // Check if this key matches the last recorded key code.
+    const auto matchingLastKeyPress = _lastVirtualKeyCode && _lastVirtualKeyCode.value() == keyEvent.GetVirtualKeyCode();

```

Comment:
cute `std::optional` trick: you don't need to check whether the optional is populated before doing a comparison; `nullopt == T{}` will always return `false`

Reply:
Oh wow, that's neat! Thank you!